core/src/main/java/io/fabric8/maven/core/service/ComposeService.java

```diff
@@ -0,0 +1,116 @@
+package io.fabric8.maven.core.service;
+
+import io.fabric8.maven.docker.util.Logger;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.shared.utils.io.FileUtils;
+import org.apache.maven.shared.utils.io.IOUtil;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.StringWriter;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+/*
+* Docker compose services for converting docker compose artifacts to kubernetes artifacts
+*/
+public class ComposeService {
+
+    public static final String KOMPOSE_RESOURCES_DIRECTORY = "kompose_resources";
+
+    private Path komposeResourcesPath;
+    private Path composeFilePath;
+    private Logger log;
+    private Process process;
+
+    /**
+     * Create instance of compose service
+     *
+     * @param composeFilePath
+     * @param log
+     */
+    public ComposeService(Path composeFilePath, Logger log) {
+        this.composeFilePath = composeFilePath;
+        this.log = log;
+    }
+
+    /**
+     * Returns array of kubernetes resource descriptor files generated from docker compose files
+     * using 'kompose' (http://kompose.io) utility
+     *
+     * @return array of files if any resources exist. The array will be empty of no resource descriptors found.
+     * @throws IOException
+     * @throws MojoExecutionException
+     */
+    public File[] convertToKubeFragments() throws IOException, MojoExecutionException {
```

Comment:
Use `Fabric8ServiceException` (or other specific exceptions) instead of `MojoExecutionException`. It's better not using Maven exceptions here.